pkg/metrics/node.go

```diff
@@ -0,0 +1,194 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/rancher/wrangler/pkg/ticker"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	v32 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	mgmtcontrollers "github.com/rancher/rancher/pkg/generated/controllers/management.cattle.io/v3"
+	v3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+)
+
+const (
+	nodeProviderLabel  = "provider"
+	nodeClusterIDLabel = "cluster_id"
+)
+
+var (
+	nodeLabels = []string{nodeProviderLabel, nodeClusterIDLabel}
```

Comment:
Clarification question - the original issue (https://github.com/rancher/rancher/issues/36410) specifies that the metrics should also expose status but i don't see us collecting that here - would that be something we want to add?

Reply:
Resolving in favor of other discussions on this topic.